src/database/schema/ChatSchema.ts

```diff
@@ -341,6 +354,29 @@ class ChatSchema implements BaseDbSchema {
     }
   };
 
+  static async deleteChat(localDb: SQLiteDatabase, messageId: string): Promise<ChatSchema> {
+    const existingChat = await ChatSchema.getByid(localDb, messageId);
+    return new ChatSchema({
+      ...existingChat,
+      type: 'deleted',
+      message: 'This message has been deleted'
+    });
+  }
+
+  static updateDeletedChatType = async (db: SQLiteDatabase, messageId: string) => {
+    try {
+      const updateQuery = `UPDATE ${ChatSchema.getTableName()}
+        SET type = ?, message = ?
+        WHERE id = ?;`;
+
+      const updateReplacement = ['deleted', 'This message has been deleted', messageId];
+
+      await db.executeSql(updateQuery, updateReplacement);
+    } catch (e) {
+      console.log('error updating deleted chat:', e);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This new function `updateDeletedChatType` updates the type and message of a chat when it's deleted. However, it doesn't handle errors properly. If an error occurs during the execution of the SQL query, it should be re-thrown after being logged, so that the caller can handle it appropriately.
```diff
    } catch (e) {
        console.log('error updating deleted chat:', e);
+       throw e;
    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->